Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm

```diff
@@ -36,8 +36,8 @@
 #import <pal/spi/cocoa/WebFilterEvaluatorSPI.h>
 #import <wtf/SoftLinking.h>
 
-SOFT_LINK_PRIVATE_FRAMEWORK(WebContentAnalysis);
-SOFT_LINK_CLASS(WebContentAnalysis, WebFilterEvaluator);
+SOFT_LINK_PRIVATE_FRAMEWORK_OPTIONAL(WebContentAnalysis);
```

Comment:
Are there no `WebContentFilterLibrary()` checks here, because `WebContentAnalysis` is only used on iOS? Should this soft linking also be restricted in that case?

Reply:
Or does the logic simply not run in places where the framework is missing?

Reply:
Never mind, I suppose `[getWebFilterEvaluatorClass() isManagedSession]` will just evaluate to false, which is fine.

Reply:
It should be fine here, since `SOFT_LINK_CLASS_OPTIONAL` will make it so that `getWebFilterEvaluatorClass()` returns `nil` (which the rest of this code handles gracefully).

However, this comment made me realize that we don't actually need to touch `ContentFilterUnblockHandler` at all, since the base system isn't a thing on iOS. I'll remove my changes there to minimize this diff.